opendbc/car/honda/carstate.py

```diff
@@ -270,38 +274,34 @@ def update(self, cp, cp_cam, _, cp_body, __) -> structs.CarState:
 
     return ret
 
-  def get_can_parser(self, CP):
-    messages = get_can_messages(CP, self.gearbox_msg)
-    return CANParser(DBC[CP.carFingerprint]["pt"], messages, CanBus(CP).pt)
+  def get_can_parsers(self, CP):
+    pt_messages = get_can_messages(CP, self.gearbox_msg)
 
-  @staticmethod
-  def get_cam_can_parser(CP):
-    messages = [
+    cam_messages = [
       ("STEERING_CONTROL", 100),
     ]
 
     if CP.carFingerprint in HONDA_BOSCH_RADARLESS:
-      messages += [
+      cam_messages += [
         ("ACC_HUD", 10),
         ("LKAS_HUD", 10),
       ]
 
     elif CP.carFingerprint not in HONDA_BOSCH:
-      messages += [
+      cam_messages += [
         ("ACC_HUD", 10),
         ("LKAS_HUD", 10),
         ("BRAKE_COMMAND", 50),
       ]
 
-    return CANParser(DBC[CP.carFingerprint]["pt"], messages, CanBus(CP).camera)
+    body_messages = [
+      ("BSM_STATUS_LEFT", 3),
+      ("BSM_STATUS_RIGHT", 3),
+    ]
+
+    return {
+      Bus.PT: CANParser(DBC[CP.carFingerprint][Bus.PT], pt_messages, CanBus(CP).pt),
+      Bus.CAM: CANParser(DBC[CP.carFingerprint][Bus.PT], cam_messages, CanBus(CP).camera),
+      Bus.BODY: CANParser(DBC[CP.carFingerprint][Bus.BODY], body_messages, CanBus(CP).radar) if CP.enableBsm else None,
```

Comment:
it simplifies things if we ensure every key has a value

Reply:
agree, this is why I had the `None` check.
guess I could rewrite as a conditional sum of dicts?